nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -443,6 +487,58 @@ public static void checkTypeParameterNullnessForAssignability(
     }
   }
 
+  private @Nullable Type inferMethodTypeArgument(
+      Type typeParam, List<Type> lhsTypeArg, List<Type> typeArg, VisitorState state) {
+    Type inferType = null;
+
+    for (int i = 0; i < typeArg.size(); i++) {
+      Type type = typeArg.get(i);
+      // base case: found type to infer to
+      if (state.getTypes().isSameType(typeParam, type)) {
+        return lhsTypeArg.get(i);
+      } else if (!type.getTypeArguments().isEmpty()) { // recursive case: generic class type
+        inferType =
+            inferMethodTypeArgument(
+                typeParam, lhsTypeArg.get(i).getTypeArguments(), type.getTypeArguments(), state);
+        if (inferType != null) {
+          return inferType;
+        }
+      }
+    }
+    // base case: typeArg doesn't contain typeParam
+    return inferType;
+  }
+
+  private Type replaceGenerics(
```

Comment:
Could this method be rewritten as a call to `com.uber.nullaway.generics.TypeSubstitutionUtils#subst`?